layers/+tools/shell/funcs.el

```diff
@@ -174,6 +174,23 @@ is achieved by adding the relevant text properties."
   (define-key eshell-mode-map
     (kbd "M-l") 'spacemacs/helm-eshell-history))
 
+(defun spacemacs/ivy-eshell-history ()
+  (interactive)
+  (counsel-esh-history)
+  ;; (evil-insert-state)
```

Comment:
Maybe this should be uncommented or removed.

`RET` doesn't run the command in normal state.

But it might be useful to insert a command from the history (for example a long path) and edit it in normal state before entering insert state and pressing `RET`.

Reply:
Right, now I tested with Evil mode. `M-l` may switch to insert state while `TAB` stays at normal state as the command name would be edited right away to add arguments. However this may be confusing. I think both should switch to insert state, similar to Helm completion.